plugins/module_utils/network/vyos/facts/ospf_interfaces/ospf_interfaces.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+# Copyright 2020 Red Hat
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+"""
+The vyos ospf_interfaces fact class
+It is in this file the configuration is collected from the device
+for a given resource, parsed, and the facts tree is populated
+based on the configuration.
+"""
+
+import re
+from copy import deepcopy
+
+from ansible_collections.ansible.netcommon.plugins.module_utils.network.common import (
+    utils,
+)
+from ansible_collections.vyos.vyos.plugins.module_utils.network.vyos.rm_templates.ospf_interfaces import (
+    Ospf_interfacesTemplate,
+)
+from ansible_collections.vyos.vyos.plugins.module_utils.network.vyos.argspec.ospf_interfaces.ospf_interfaces import (
+    Ospf_interfacesArgs,
+)
+
+
+class Ospf_interfacesFacts(object):
+    """The vyos ospf_interfaces facts class"""
+
+    def __init__(self, module, subspec="config", options="options"):
+        self._module = module
+        self.argument_spec = Ospf_interfacesArgs.argument_spec
+        spec = deepcopy(self.argument_spec)
+        if subspec:
```

Comment:
I think this section of code (37-46) is not required. I will remove it from builder template. But for now, removing it from here will bump code cov too.

Reply:
I might have to change this assertion for most of the interfaces related tests. I shall take it up as a separate PR.

Reply:
Okay